packages/terra-theme-context/src/terra-dev-site/doc/ThemeContext/ThemeContext.1.doc.mdx

```diff
@@ -0,0 +1,44 @@
+import { Badge } from 'terra-theme-context/package.json?dev-site-package';
+
+import ThemeContextProviderExample from '../example/ThemeContextProviderExample?dev-site-example';
+import ThemedComponent from '../example/ThemedComponent?dev-site-example';
+
+<Badge />
+
+# Terra Theme Context
+A react context to provide the current theme. This package is intended to be used by terra components only, Any terra consumers or custom terra components should consume the context from the terra-application package.
+
+This context can be used to apply any theme related changes to a component.
+
+The most common use would be to apply a theme class to the root tag of the component to apply theme variables. See below for an example.
```

Comment:
There's two different things I think may trip people up that might be worth a few notes.

1. That a default theme is no theme within the context. `<ThemeContext.Provider value={{ theme: '' }}>`
2. The differences between the newly added `ThemeContext/Provider` in this PR and the existing [ThemeProvider](https://engineering.cerner.com/terra-ui/components/terra-theme-provider/theme-provider/theme-provider) and when to use which one.

Reply:
I was imagining theme would be set to undefined when default. Do you have any suggestions or concerns about that? It feels a little weird to be honest. I could default it to 'default'...

I'll update the documentation here, but really I DO NOT want to add this package to terra-ui and would rather update terra-application docs to provide how people should consume the theme-context.

Reply:
It does feel weird. Setting it to `default` will add an unbound class to each of the terra components assuming we pipe the theme directly into `cx()`. Unless we validate it first, add a default class, or abstract the default theme out into a theme rather than a fallback. Undefined and empty string are also weird.

Reply:
Since both options feel weird, for now i'm sticking with default theme = undefined for the benefit of being able to directly pass it to cx and not apply any classes.

Reply:
Doc updated: https://github.com/cerner/terra-framework/pull/1087/commits/8398f7e62234b6c44502add564933a56987e3a40